frontend/app/routes/_gcweb-app.personal-information.preferred-language.edit.tsx

```diff
@@ -47,6 +50,8 @@ export async function action({ request }: ActionFunctionArgs) {
   const session = await sessionService.getSession(request.headers.get('Cookie'));
   session.set('preferredLanguage', parsedDataResult.data.preferredLanguage);
 
+  const userId = await userService.getUserId();
+  await userService.updateUserInfo(userId, parsedDataResult.data);
```

Comment:
This is not required here since `confirm` action route has been added and will update the preferred language.

Reply:
The `confirm` route has been added last week to have a consistent global edit flow. That route wasn't there at the moment of our discussions.